include/ppx/geometry.h

```diff
@@ -157,43 +167,68 @@ class Geometry
         char*       GetData() { return DataPtr(mData); }
         const char* GetData() const { return DataPtr(mData); }
         uint32_t    GetElementCount() const;
+        uint32_t    GetDataSize() const;
 
         // Trusts that calling code is well behaved :)
         //
+        // Write(count, pValues...) is used to write a chunk of data at specified offset
+        // If called with offset=-1, will write at the end and adjust mOffset accordingly
```

Comment:
mOffset is a private field, and this is a public function. I think it should be more verbose about what happens to offset. For example "updates internal offset that points to the end of the written data" or something like that.